pkg/metrics/server.go

```diff
@@ -37,14 +44,77 @@ func RunServer(port int, stopCh <-chan struct{}) {
 		Handler: router,
 	}
 
-	go func() {
-		if err := srv.ListenAndServeTLS(tlsCRT, tlsKey); err != nil {
-			klog.Errorf("error starting metrics server: %v", err)
+	return srv
+}
+
+func startServer(srv *http.Server, stopCh <-chan struct{}) {
+	klog.Infof("starting metrics server")
+	if err := srv.ListenAndServeTLS(tlsCRT, tlsKey); err != nil && err != http.ErrServerClosed {
+		klog.Errorf("error from metrics server: %v", err)
+	}
+}
+
+func stopServer(srv *http.Server) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := srv.Shutdown(ctx); err != nil {
+		klog.Errorf("error or timeout stopping metrics listener: %v", err)
+	} else {
+		// TODO delete debugging lines
```

Comment:
I think that comment is obsolete.

Reply:
I think this line and the line below is the debugging that should be removed.  Anyways, still WiP and will be cleaned I believe.